src/components/Tool/FixSendNewsletter/FixSendNewsletter.tsx

```diff
@@ -143,110 +110,122 @@ const FixSendNewsletter: React.FC<Props> = ({
+                    }}
+                  />
+                }
+              </strong>
+            </Typography>
+            <Typography>
+              {t(
+                'Contacts that appear here have an empty Newsletter Status and Partner Status set to Financial, Special, or Pray. Choose a newsletter status for contacts below.',
+              )}
+            </Typography>
+          </Box>
+        </Grid>
+        {!loading && data && !!numberOfContacts ? (
+          <>
+            <ButtonHeaderBox mb={0}>
+              <Box>
+                <Typography>
+                  <Trans
+                    defaults="<i>Showing <bold>{{numberOfContacts}}</bold> of <bold>{{totalCount}}</bold></i>"
+                    shouldUnescape
+                    values={{
+                      numberOfContacts,
+                      totalCount,
+                    }}
+                    components={{ bold: <strong />, i: <i /> }}
+                  />
+                </Typography>
+              </Box>
+              {(loading || updating) && <LoadingSpinner firstLoad={true} />}
+              <Box>
+                <Button
+                  variant="contained"
+                  onClick={() => null}
+                  disabled={updating || !numberOfContacts}
+                  sx={{ mr: 2 }}
+                >
+                  {
                     <Trans
-                      defaults="Showing <bold>{{value}}</bold> of <bold>{{value}}</bold>"
-                      shouldUnescape
+                      defaults="Confirm All ({{value}})"
                       values={{
                         value: numberOfContacts,
                       }}
-                      components={{ bold: <strong /> }}
                     />
-                  </Typography>
-                </Box>
-              </Grid>
-            </>
-          ) : (
-            <NoData tool="fixSendNewsletter" />
-          )}
-        </Grid>
-      ) : (
-        <CircularProgress style={{ marginTop: theme.spacing(3) }} />
-      )}
+                  }
+                </Button>
+              </Box>
+            </ButtonHeaderBox>
+            {contactsToFix?.map((contact) => (
+              <Contact
+                id={contact.id}
+                name={contact.name}
+                // need to fix this after changes to fix commitment info get merged
+                avatar={contact.avatar}
+                status={
+                  data.constant.status?.find(
+                    (status) => contact.status === status.id,
+                  )?.value || ''
+                }
+                primaryPerson={
+                  contact.primaryPerson || {
+                    id: '',
+                    firstName: '',
+                    lastName: '',
+                    primaryEmailAddress: {
+                      email: '',
+                      id: '',
+                    },
+                    optoutEnewsletter: false,
```

Comment:
I feel like you could create an object outside of this component and use that as the defaults, for this and `primaryAddress`